client/state/data-getters/index.js

```diff
@@ -142,31 +142,39 @@ export const requestSiteAlerts = siteId => {
 	);
 };
 
-export const requestGutenbergDraftPost = ( siteId, draftId ) =>
+export const createAutoDraft = ( siteId, draftKey, postType ) =>
 	requestHttpData(
-		draftId,
+		draftKey,
 		http(
 			{
-				path: `/sites/${ siteId }/posts/auto-draft`,
+				path: `/sites/${ siteId }/posts/new`,
 				method: 'POST',
-				apiNamespace: 'wpcom/v2',
-				body: {}, //this is for a POST verb.
+				apiVersion: '1.2',
+				body: {
+					status: 'auto-draft',
+					type: postType,
+					content: ' ', //endpoint only creates this with non-empty content ¯\_(ツ)_/¯
+				},
 			},
 			{}
 		),
-		{ fromApi: () => data => [ [ draftId, data ] ] }
+		{ fromApi: () => data => [ [ draftKey, data ] ] }
 	);
 
-export const requestSitePost = ( siteId, postId ) =>
-	requestHttpData(
+export const requestSitePost = ( siteId, postId, postType ) => {
+	//post and page types are plural except for custom post types
+	//eg /sites/<siteId>/posts/1234 vs /sites/<siteId>/jetpack-testimonial/4
+	const s = postType === 'page' || postType === 'post' ? 's' : '';
```

Comment:
This is a clever one, but I'm personally not a big fan of adding the plural `s` this way, and I'd rather have a more straightforward if/else-if/else, with the additional advantage of being self-explanatory.